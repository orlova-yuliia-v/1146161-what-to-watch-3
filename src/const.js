export const ALL_GENRES = `All genres`;

export const DEFAULT_SHOWED_MOVIES_NUMBER = 8;

export const MAX_SIMILAR_MOVIES_NUMBER = 4;

export const MAX_GENRES_NUMBER = 10;

export const ReviewLength = {
  MIN: 50,
  MAX: 400
};

export const TabName = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  ADD_REVIEW: `/review`,
  MY_LIST: `/mylist`,
  FILMS: `/films`,
  PLAYER: `/player`
};
