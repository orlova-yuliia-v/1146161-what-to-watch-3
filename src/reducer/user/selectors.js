import NameSpace from "../name-space.js";

export const getAuthorizationStatus = (state) => {
  return state[NameSpace.USER].authorizationStatus;
};

export const getAuthUser = (state) => {
  return state[NameSpace.USER].authUserData;
};
