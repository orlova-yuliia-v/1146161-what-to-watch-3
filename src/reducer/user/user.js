import {extend, normalizeUserData} from "../../utils.js";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  authUserData: {}
};

const ActionType = {
  REQUIRE_AUTHORIZATION: `REQUIRE_AUTHORIZATION`,
  SET_AUTH_USER_DATA: `SET_AUTH_USER_DATA`
};

const Operation = {
  checkAuth: () => (dispatch, _getState, api) => {

    return api
    .get(`/login`)
      .then((response) => {
        if (!response) {
          return;
        }
        const normalizedAuthData = normalizeUserData(response.data);
        dispatch(ActionCreator.setAuthUserData(normalizedAuthData));
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      })

      .catch((err) => {
        throw err;
      });
  },
  login: (authData) => (dispatch, _getState, api) => {
    return api
      .post(`/login`, {
        email: authData.login,
        password: authData.password
      })
      .then((response) => {
        if (!response) {
          return;
        }
        const normalizedAuthData = normalizeUserData(response.data);
        dispatch(ActionCreator.setAuthUserData(normalizedAuthData));
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      })

      .catch((err) => {
        throw err;
      });
  }
};

const ActionCreator = {
  requireAuthorization: (status) => {
    return {
      type: ActionType.REQUIRE_AUTHORIZATION,
      payload: status
    };
  },
  setAuthUserData: (authUserData) => {
    return {
      type: ActionType.SET_AUTH_USER_DATA,
      payload: authUserData
    };
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRE_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload
      });
    case ActionType.SET_AUTH_USER_DATA:
      return extend(state, {
        authUserData: action.payload
      });
  }

  return state;
};

export {reducer, ActionCreator, ActionType, AuthorizationStatus, Operation};
