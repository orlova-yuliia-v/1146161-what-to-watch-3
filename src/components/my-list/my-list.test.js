import React from "react";
import renderer from "react-test-renderer";
import MyList from "./my-list.jsx";
import configureStore from "redux-mock-store";
import {ALL_GENRES, DEFAULT_SHOWED_MOVIES_NUMBER} from "../../const.js";
import NameSpace from "../../reducer/name-space.js";
import {AuthorizationStatus} from "../../reducer/user/user.js";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";

const mockStore = configureStore([]);

const authDataMock = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};

const movies = [{
  id: 1,
  title: `Some title`,
  poster: `1.jpg`,
  bgPosterUrl: `https://image-url.com/1.jpg`,
  genre: `Some genre`,
  releaseYear: 2020,
  director: `Director name`,
  actors: [`Actor 1`, `Actor 2`, `Actor 3`, `Actor 4`, `Actor 5`],
  runTime: 2,
  ratingScore: 7.5,
  ratingCount: 1234,
  description: `Film description`,
  previewUrl: `https://preview-url.com/1.mp4`,
  reviews: [
    {
      ratingScore: 8.5,
      date: `September 8, 2019`,
      author: `Yuliia Orlova`,
      text: `Review text`
    }
  ]
},
{
  id: 2,
  title: `Some title`,
  poster: `1.jpg`,
  bgPosterUrl: `https://image-url.com/1.jpg`,
  genre: `Some genre`,
  releaseYear: 2020,
  director: `Director name`,
  actors: [`Actor 1`, `Actor 2`, `Actor 3`, `Actor 4`, `Actor 5`],
  runTime: 2,
  ratingScore: 7.5,
  ratingCount: 1234,
  description: `Film description`,
  previewUrl: `https://preview-url.com/1.mp4`,
  reviews: [
    {
      ratingScore: 8.5,
      date: `September 8, 2019`,
      author: `Yuliia Orlova`,
      text: `Review text`
    }
  ]
}];

it(`Should render MyList component`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      movies,
      promoMovie: movies[0],
      movieComments: []
    },
    [NameSpace.STATE]: {
      selectedGenre: ALL_GENRES,
      showedMovies: DEFAULT_SHOWED_MOVIES_NUMBER,
      isFullVideoPlayerVisible: false,
      selectedMovieId: 1
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      authUserData: {authDataMock}
    }
  });

  const tree = renderer
    .create(
        <MemoryRouter>
          <Provider store={store}>
            <MyList
              movies={movies}
              onMovieCardClick={() => { }}
              authUserData={authDataMock} />
          </Provider>
        </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
