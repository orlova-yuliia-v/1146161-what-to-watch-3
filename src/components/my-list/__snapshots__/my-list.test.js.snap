// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MyList component 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            height="63"
            src="https://htmlacademy-react-3.appspot.com/undefined"
            width="63"
          />
        </div>
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            height="175"
            src="1.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
          >
            Some title
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            height="175"
            src="1.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
          >
            Some title
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
